subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -817,6 +817,33 @@ static void pdu_free_sem_give(void)
 }
 #endif /* !CONFIG_BT_CTLR_ZLI */
 
+static void pdu_free_sem_take_all_no_wait(void)
+{
+	int err;
+
+	/* Flush all given semaphore count due to multiple release from LLL
+	 * context.
+	 */
+	do {
+		err = k_sem_take(&sem_pdu_free, K_NO_WAIT);
```

Comment:
Why not `k_sem_reset()` ?

Reply:
Was not aware of this! ... Let @caco3 verify the current fix, I will replace it with `k_sem_reset()` and verify again.